application/common/components/helpers/Utils.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+namespace Sil\Idp\IdSync\common\components\helpers;
+
+class Utils
```

Comment:
A class named "Utils" can often be a code smell. Is there a way you could make this class more specific to the problem you're solving with it? What would you think of renaming it to something like "ArrayCollapser", with a "collapseRecursively()" method?

There might not be any simple, better way to do it, in which case what you have here will work fine. It's just so easy for a "Utils" class to become a messy collection of unrelated functions that it seemed worth asking if there's a better way in this case.

Reply:
I see what you mean, though it's hard to forsee what else might be needed since there's only one method currently. Would it be better to just pull it back into `SagePeopleIdStore` since that's the only place it's used?

Reply:
No, I'd recommend keeping it in a separate class like this. If another need comes up in the future, it will either line up with the purpose of the new class (e.g. "ArrayCollapser"), in which case it can be added, or it won't, in which case it should be in a separate class anyway.

Basically, I'm trying to think of this in light of the idea that each class should only have one reason to change (or phrased a different way, should only have a single responsibility).

Reply:
Ok, I'll keep it separate. (To be continued on PR #88)